internal/controller/controllers/controller_event_wrapper.go

```diff
@@ -26,18 +26,18 @@ func NewControllerEventsWrapper(crdEventsHandler CRDEventsHandler, events events
 		events: events, db: db, log: log}
 }
 
-func (c *controllerEventsWrapper) AddEvent(ctx context.Context, clusterID strfmt.UUID, hostID *strfmt.UUID, severity string, msg string, eventTime time.Time, props ...interface{}) {
-	c.events.AddEvent(ctx, clusterID, hostID, severity, msg, eventTime, props)
+func (c *controllerEventsWrapper) V2AddEvent(ctx context.Context, clusterID *strfmt.UUID, hostID *strfmt.UUID, infraEnvID *strfmt.UUID, severity string, msg string, eventTime time.Time, props ...interface{}) {
+	c.events.V2AddEvent(ctx, clusterID, hostID, infraEnvID, severity, msg, eventTime, props)
 
 	if hostID != nil {
-		c.NotifyKubeApiHostEvent(clusterID, *hostID)
+		c.NotifyKubeApiHostEvent(common.StrFmtUUIDVal(infraEnvID), common.StrFmtUUIDVal(hostID))
 	} else {
-		c.NotifyKubeApiClusterEvent(clusterID)
+		c.NotifyKubeApiClusterEvent(common.StrFmtUUIDVal(clusterID))
```

Comment:
not related to this pr, but what will happen if cluster id is nil?

Reply:
I'll double-check.

Reply:
Per our conversation, I've added a check for this case in `NotifyKubeApiClusterEvent`, and also in `NotifyKubeApiHostEvent` and `NotifyKubeApiInfraEnvEvent`.

In addition, there's see a unit test in `controller_event_wrapper_test.go` that does send a nil clusterID which I modified in this PR named: Adding a host event